File: kargerMinCut/golang/kargerMinCut_test.go
package kargerMinCut

import (
	"testing"
	"github.com/stretchr/testify/assert"
	//"fmt"
)

func TestKargerMinCut(t *testing.T) {
	testCases := []struct{
		inputEdges [][]int
		expectMinCut int
	} {
		{
			[][]int{},
			0,
		},
		{
			nil,
			0,
		},
		{
			[][]int{
				[]int{0,1},
				[]int{1,2},
				[]int{2,3},
				[]int{0,4},
				[]int{1,5},
				[]int{2,6},
				[]int{3,7},
				[]int{4,5},
				[]int{5,6},
				[]int{6,7},
				[]int{0,5},
				[]int{1,4},
				[]int{2,7},
				[]int{3,6},
			},
			2,
		},
		{
			[][]int{
				[]int{0,1},
				[]int{0,2},
				[]int{0,3},
				[]int{0,4},
				[]int{1,2},
				[]int{1,3},
				[]int{1,4},
				[]int{2,3},
				[]int{2,4},
				[]int{3,4},
				[]int{5,6},
				[]int{5,7},
				[]int{5,8},
				[]int{5,9},
				[]int{6,7},
				[]int{6,8},
				[]int{6,9},
				[]int{7,8},
				[]int{7,9},
				[]int{8,9},
				[]int{0,5},
				[]int{2,6},
				[]int{4,7},
			},
			3,
		},
	}

	for _, testCase := range testCases {
		acutalMinCut := doMinCut(testCase.inputEdges)
		assert.Equal(t, testCase.expectMinCut, acutalMinCut)
	}
}

File: kargerMinCut/python3/karger_mincut_test.py
import unittest
import karger_mincut

class KargerMinCutTestCases(unittest.TestCase):
    def test_do_min_cut(self):
        test_cases = [
            {
                'input': None,
                'expect': 0 
            },
            {
                'input': [],
                'expect': 0 
            },
            {
                'input':  [
                    [1,2],
                    [2,3],
                    [0,4],
                    [1,5],
                    [2,6],
                    [3,7],
                    [4,5],
                    [5,6],
                    [6,7],
                    [0,5],
                    [1,4],
                    [2,7],
                    [3,6],
                ],
                'expect': 2
            },
            {
                'input':  [
                    [0,1],
                    [0,2],
                    [0,3],
                    [0,4],
                    [1,2],
                    [1,3],
                    [1,4],
                    [2,3],
                    [2,4],
                    [3,4],
                    [5,6],
                    [5,7],
                    [5,8],
                    [5,9],
                    [6,7],
                    [6,8],
                    [6,9],
                    [7,8],
                    [7,9],
                    [8,9],
                    [0,5],
                    [2,6],
                    [4,7]
                ],
                'expect': 3
            }]

        for test_case in test_cases:
            actual_min_cut = karger_mincut.do_min_cut(test_case['input'])
            self.assertEqual(test_case['expect'], actual_min_cut)

if __name__ == '__main__':unittest.main()
